File: lana/src/main/scala/com/financialforce/lana/display/WebView.scala
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */
package com.financialforce.lana.display

import com.financialforce.lana.runtime.vscode.{Uri, WebviewPanel, WebviewPanelOptions, window}

import scala.scalajs.js

class WebViewOptions(resourceRoots: Seq[Uri]) extends WebviewPanelOptions {
  override val enableCommandUris: Boolean = true
  override val enableScripts: Boolean = true
  override val retainContextWhenHidden: Boolean = true
  override val localResourceRoots: js.Array[Uri] = js.Array(resourceRoots: _*)
  override val enableFindWidget: Boolean = true
}

object WebView {
  def apply(name: String, title: String, resourceRoots: Seq[Uri]): WebviewPanel =
    window.createWebviewPanel(name, title, -1, new WebViewOptions(resourceRoots))
}

File: src/display/WebView.ts
/*
 * Copyright (c) 2020 FinancialForce.com, inc. All rights reserved.
 */

import { Uri, WebviewPanel, WebviewPanelOptions, window } from "vscode";

export class WebViewOptions implements WebviewPanelOptions {
  enableCommandUris: boolean = true;
  enableScripts: boolean = true;
  retainContextWhenHidden: boolean = true;
  localResourceRoots: Uri[];
  enableFindWidget: boolean = true;

  constructor(resourceRoots: Uri[]) {
    this.localResourceRoots = resourceRoots;
  }
}

export class WebView {
  static apply(
    name: string,
    title: string,
    resourceRoots: Uri[]
  ): WebviewPanel {
    return window.createWebviewPanel(
      name,
      title,
      -1,
      new WebViewOptions(resourceRoots)
    );
  }
}
